hw2/aes_power_analysis.py

```diff
@@ -64,7 +64,7 @@ def simulate_power_analysis(plaintexts, key, start_round, end_round):
         for r in range(max(start_round, 1), end_round + 1):
             stateround = aes.encrypt_r(plaintexts[j], r, start_round)
             for i in range(BlockSize):
-                ?
+                hamming_distances[j][r - 1][i] = hamming_weight(aes.S[stateround[i]] ^ stateround[i])

```

Comment:
Does not seem very crucial but should have put `r` and not `r-1`, as described in return value